server/src/main/java/com/epam/aidial/core/server/service/ShareService.java

```diff
@@ -105,12 +111,33 @@ public SharedResourcesResponse listSharedByMe(String bucket, String location, Li
         return new SharedResourcesResponse(resultMetadata);
     }
 
+
+    private void addCustomApplicationRelatedFiles(ShareResourcesRequest request) {
+        List<String> filesFromRequest = request.getResources().stream()
+                .map(SharedResource::url).toList();
+        Config config = configStore.load();
+        Set<SharedResource> newSharedResources = new HashSet<>(request.getResources());
+        for (SharedResource sharedResource : request.getResources()) {
+            ResourceDescriptor resource = getResourceFromLink(sharedResource.url());
+            if (resource.getType() == ResourceTypes.APPLICATION) {
+                Application application = applicationService.getApplication(resource).getValue();
+                List<ResourceDescriptor> files = ApplicationTypeSchemaUtils.getFiles(config, application, encryptionService, resourceService);
+                for (ResourceDescriptor file : files) {
+                    if (!filesFromRequest.contains(file.getUrl())) {
```

Comment:
You should skip public and not-your-bucket resources. Please write a test to cover this logic. An app has a public or/and a shared resource. The invitation should not include these files.

Reply:
I propose to throw invalid op that case

Reply:
done